### ISSUE TITLE
Sync detects file changes but does not properly apply them

### ISSUE THREAD

**Issue by ericfiegel** [GoogleContainerTools/skaffold#7428]
### Expected behavior
proper file sync to a running python app (FastAPI)

### Actual behavior
```
INFO[0211] files modified: [my-service/api/__init__.py]  subtask=-1 task=DevLoop
DEBU[0211]  devloop: build true, sync true, deploy true  subtask=-1 task=DevLoop
Syncing 1 files for my-service:6be91a4e80715cc31e43f5f18cd7f70183b1d0aa4473034ba5beb6d8dc5ec936
INFO[0211] Copying files:map[my-service/api/__init__.py:[/usr/src/app/my-service/api/__init__.py]]tomy-service:6be91a4e80715cc31e43f5f18cd7f70183b1d0aa4473034ba5beb6d8dc5ec936  subtask=-1 task=DevLoop
DEBU[0211] getting client config for kubeContext: `minikube`  subtask=-1 task=DevLoop
DEBU[0211] Running command: [kubectl --context minikube exec my-service-6c74976fff-2rdzd --namespace default -c my-service -i -- tar xmf - -C / --no-same-owner]  subtask=-1 task=DevLoop
DEBU[0211] Command output: [], stderr: tar: Removing leading `/' from member names  subtask=-1 task=DevLoop
```

service redeploys but without updates. exec into pod and file changes aren't there.

### Information

- Skaffold version: v1.36.1
- Operating system: macOS Big Sur 11.6.2 
- Installed via: Homebrew 
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v2beta27
kind: Config
metadata:
  name: my-service
build:
  local:
    push: false
  artifacts:
    - image: my-service
      context: .
      docker:
        dockerfile: Dockerfile
      sync:
        manual:
          - src: "**/*.py"
            dest: .
deploy:
  helm:
    releases:
      - name: my-service
        artifactOverrides:
          deployment.image: my-service
        imageStrategy:
          helm: {}
        chartPath: helm
        valuesFiles:
          - values-local.yaml
portForward:
  - resourceType: deployment
    resourceName: my-service
    port: 9000
    localPort: 9000
```

- Contents of Dockerfile:
```
FROM python:3.9-buster

RUN apt-get update

RUN apt-get install libcurl4-openssl-dev

COPY config/pip.conf  root/.pip/pip.conf
COPY Pipfile .
COPY Pipfile.lock .

RUN pip install pipenv

RUN pipenv install --system --ignore-pipfile

WORKDIR /usr/src/app/

ENTRYPOINT ["bin/entrypoint.sh"]

COPY . /usr/src/app
```

### Steps to reproduce the behavior

1. `skaffold dev -v=debug`
2. change and save a file in the repo
3. updates not applied

some debug logs
```
DEBU[0205] Change detectednotify.Write: "<home dir>/my-service/api/__init__.py~"  subtask=-1 task=DevLoop
DEBU[0205] Change detectednotify.Write: "<home dir>/my-service/api/__init__.py~"  subtask=-1 task=DevLoop
DEBU[0205] Change detectednotify.Create: "<home dir>/my-service/api/__init__.py~"  subtask=-1 task=DevLoop
DEBU[0205] Change detectednotify.Create: "<home dir>/my-service/api/__init__.py~"  subtask=-1 task=DevLoop
DEBU[0205] Change detectednotify.Write: "<home dir>/my-service/api/__init__.py"  subtask=-1 task=DevLoop
DEBU[0205] Change detectednotify.Write: "<home dir>/my-service/api/__init__.py"  subtask=-1 task=DevLoop
DEBU[0205] Change detectednotify.Remove: "<home dir>/my-service/api/__init__.py~"  subtask=-1 task=DevLoop
DEBU[0205] Change detectednotify.Remove: "<home dir>/my-service/api/__init__.py~"  subtask=-1 task=DevLoop
```

```
INFO[0210] files modified: [<home dir>/my-service/api/__init__.py]  subtask=-1 task=DevLoop
```

```
INFO[0211] files modified: [my-service/api/__init__.py]  subtask=-1 task=DevLoop
DEBU[0211]  devloop: build true, sync true, deploy true  subtask=-1 task=DevLoop
Syncing 1 files for my-service:6be91a4e80715cc31e43f5f18cd7f70183b1d0aa4473034ba5beb6d8dc5ec936
INFO[0211] Copying files:map[my-service/api/__init__.py:[/usr/src/app/my-service/api/__init__.py]]tomy-service:6be91a4e80715cc31e43f5f18cd7f70183b1d0aa4473034ba5beb6d8dc5ec936  subtask=-1 task=DevLoop
DEBU[0211] getting client config for kubeContext: `minikube`  subtask=-1 task=DevLoop
DEBU[0211] Running command: [kubectl --context minikube exec my-service-6c74976fff-2rdzd --namespace default -c my-service -i -- tar xmf - -C / --no-same-owner]  subtask=-1 task=DevLoop
DEBU[0211] Command output: [], stderr: tar: Removing leading `/' from member names  subtask=-1 task=DevLoop
```

**Comment by zeno-mioso** [comment]
same behavior also for me, with k3d intesad of minikube and using kaniko builder.

File change is detected but propagation to the running pod silently fails

**Comment by zeno-mioso** [comment]
this is my debug level output:


```
DEBU[2769] Change detectednotify.Write: "/MY/MACHINE/PATH/core/app/Models/Role.php"  subtask=-1 task=DevLoop
DEBU[2769] Change detectednotify.Write: "/MY/MACHINE/PATH/core/app/Models/Role.php"  subtask=-1 task=DevLoop
DEBU[2770] Found dependencies for dockerfile: [{package-lock.json /app true 7 7} {package.json /app true 7 7} {package-lock.json /app true 7 7} {package.json /app true 7 7} {. /app true 16 16} {. /app true 16 16}]  subtask=-1 task=DevLoop
DEBU[2770] Found dependencies for dockerfile: [{composer.json /code true 27 27} {composer.lock /code true 27 27} {.php.local.ini /usr/local/etc/php/conf.d/local.ini false 62 62} {. /code true 63 63} {.docker/supervisord/queue-worker.ini /etc/supervisor.d/queue-worker.ini false 74 74}]  subtask=-1 task=DevLoop
INFO[2770] files modified: [core/app/Models/Role.php]    subtask=-1 task=DevLoop
DEBU[2770] Found dependencies for dockerfile: [{.php.local.ini /usr/local/etc/php/conf.d/local.ini false 62 62} {. /code true 63 63} {.docker/supervisord/queue-worker.ini /etc/supervisor.d/queue-worker.ini false 74 74}]  subtask=-1 task=DevLoop
DEBU[2770]  devloop: build false, sync true, deploy false  subtask=-1 task=DevLoop
Syncing 1 files for MYREMOTEREGISTRY:5000/core:latest@sha256:589c1a6c606f1b1f27cb4c4a0f0d7274795e99ed725942520a52cf0dada60fe0
INFO[2770] Copying files:map[core/app/Models/Role.php:[/code/app/Models/Role.php]]toMYREMOTEREGISTRY:5000/core:latest@sha256:589c1a6c606f1b1f27cb4c4a0f0d7274795e99ed725942520a52cf0dada60fe0  subtask=-1 task=DevLoop
DEBU[2770] getting client config for kubeContext: `k3d-CLUSTER`  subtask=-1 task=DevLoop
INFO[2770] Copying files:map[core/app/Models/Role.php:[/code/app/Models/Role.php]]toMYREMOTEREGISTRY:5000/core:latest@sha256:589c1a6c606f1b1f27cb4c4a0f0d7274795e99ed725942520a52cf0dada60fe0  subtask=-1 task=DevLoop
DEBU[2770] getting client config for kubeContext: `k3d-CLUSTER`  subtask=-1 task=DevLoop
Watching for changes...
```


I tryied to run kubectl cp util to simulate the correct behavior and file get pushed correctly, without errors.

**Comment by ericfiegel** [comment]
I finally figured out what happened here! Seems to actually be working just fine. I had another service (a database) that was triggering rebuilds often, and those were forcing all of my services to redeploy. I ignored changes on the database service and this file sync worked as expected! It was always working, but the synced changes were being then overwritten by a redeploy of the old image when the other service was triggered.
